### PR TITLE
ERC-4786: Link Common Token to ERC-721

### PR DESCRIPTION
**Pull request by poria-cat**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-4786.md [hunk 1]
```diff
@@ -0,0 +1,403 @@
+---
+eip: 4786
+title: Link Common Token to ERC-721
+description: Link ERC721 token with ERC721/ERC1155/ERC20 token
```
Comment (lightclient) [inline EIPS/eip-4786.md:4]:
```suggestion
description: Link ERC-721 token with ERC-721/ERC-1155/ERC-20 token
```

File: EIPS/eip-4786.md [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+requires: 165
+---
+
+## Abstract
+ERC-4786 provides an extension for ERC-721 to be composed with other Tokens (ERC-721/ERC-1155/ERC-20). This applies to the creation of a composable/graph NFT. In this standard, the ERC-721 Token is a first class citizen and can freely compose with other Tokens. The result is get an interesting NFT with a graph structure.
+
+## Motivation
+The ability to compose opens up many more possibilities for ERC-721, here are some possible scenarios:
+
+1. assets owned by an address can be transformed into assets owned by an NFT.
```
Comment (lightclient) [inline EIPS/eip-4786.md:20]: Please start all points with a capital letter.

**Review by lightclient — CHANGES_REQUESTED** [review]

File: EIPS/eip-4786.md [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+
+- [Poria Cattus ERC-4786 reference implementation](https://github.com/poria-cat/ERC-4786)
+
+## Security Consideration
+There are no security considerations related directly to the implementation of this standard.
+
+## Copyright
+Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
+
+## Citation
```
[the remark this answers is not in the document]
Comment (poria-cat) [reply EIPS/eip-4786.md:393]: copyright removed :hugs:

File: EIPS/eip-4786.md [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+
+If the contract code for ERC-4786 and ERC-721 is placed directly in one contract, this may cause the contract to exceed the size limit, so it is better to deploy it in two separate contracts. If you want to use a concept similar to the container NFT token, you can limit the targetToken of the link to only the NFTs minted in a particular contract  and check the target token of the updateTarget.
+
+## Backwards Compatibility
+The ERC-4786 is an extension of the ERC-721 and does not modify the interface of ERC-721, so it is 100% compatible with the ERC-721.
+
+## Test Cases
+Poria Cattus's ERC-4786 reference implementation includes test cases written using Hardhat.
+
+## References
```
[the remark this answers is not in the document]
Comment (poria-cat) [reply EIPS/eip-4786.md:375]: Test case removed :hugs:

File: EIPS/eip-4786.md [hunk 5]
```diff
@@ -0,0 +1,397 @@
...
+- [Poria Cattus ERC-4786 reference implementation](https://github.com/poria-cat/ERC-4786)
+
+## Security Consideration
+There are no security considerations related directly to the implementation of this standard.
+
+## Citation
+
+Please cite this document as:
+
+[Poria Cattus](https://github.com/poria-cat), "EIP-4786: Link Common Token to ERC-721," *Ethereum Improvement Proposals*, no. 4786, February 2022. [Online serial]. Available: https://eips.ethereum.org/EIPS/eip-4786.
```
Comment (MicahZoltu) [inline EIPS/eip-4786.md:397]:
```suggestion
## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```
Please see https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for an example of the proper layout of an EIP.

Comment (poria-cat) [reply EIPS/eip-4786.md]: Thanks @MicahZoltu , I get it. I also changed the Reference Implementation section.

**Review by lightclient — CHANGES_REQUESTED** [review]
`s/Emited/Emitted/g`

**Pull request closed without merging**